qiita_db/processing_job.py

```diff
@@ -880,12 +880,30 @@ def submit(self, parent_job_id=None, dependent_jobs_list=None):
         # portal server that submitted the job
         url = "%s%s" % (qiita_config.base_url, qiita_config.portal_dir)
 
+        # if the word ENVIRONMENT is in the plugin_env_script we have a special
+        # case where we are going to execute some command and then wait for the
+        # plugin to return their own id (first implemented for
+        # fast-bowtie2+woltka)
+        if 'ENVIRONMENT' in plugin_env_script:
+            # the job has to be in running state so the plugin can change its`
+            # status
+            with qdb.sql_connection.TRN:
+                self._set_status('running')
+                qdb.sql_connection.TRN.commit()
+
+            create_nested_path(job_dir)
+            cmd = (f'{plugin_env_script}; {plugin_start_script} '
+                   f'{url} {self.id} {job_dir}')
+            stdout, stderr, return_value = _system_call(cmd)
+            if return_value != 0 or stderr != '':
+                self._set_error(stderr)
+            job_id = stdout
```

Comment:
Does this string need to be stripped for new lines or something like that?

Reply:
Excellent question! IMOO no. Basically, the idea is to display whatever information the process creating the job returns.

Reply:
Got it, should be used to display to the end-user not to decode a return value. Makes sense.